projects/buttons_and_adcs/src/main.c

```diff
@@ -0,0 +1,58 @@
+// Create a project that reads an ADC converted reading from pin A6 whenever a button connected to
+// pin B2 and logs the output. The adc reading should trigger whenever the button is released, not
+// initially when the button is pressed. Follow the same submission procedures as the firmware 102
+// homework, except the suffix of the branch should be fw_103.
+
+// Hints:
+// assume the button is active-high
+// register an interrupt with the appropriate InterruptEdge settings (google this or ask someone /
+// your lead)
+#include <stdbool.h>
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "interrupt_def.h"
+#include "log.h"
+
+void prv_button_interrupt_handler(const GpioAddress *address, void *context) {
+  uint16_t adc_pin_data = 0;
+  GpioAddress *adc_pin_addr = (GpioAddress *)context;
+  adc_read_raw_pin(*adc_pin_addr, &adc_pin_data);
```

Comment:
The homework does specify a *converted* reading: that means use `adc_read_converted_pin` instead of `adc_read_raw_pin`. The difference is that `adc_read_converted_pin` returns a value in millivolts, whereas `adc_read_raw_pin` returns whatever raw value the ADC spits out - you almost always want a converted read.